lab_5_scrapper/scrapper.py

```diff
@@ -120,45 +263,126 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
         """
         Initializes an instance of the HTMLParser class
         """
-        pass
+        self.full_url = full_url
+        self.article_id = article_id
+        self.config = config
+        self.article = Article(self.full_url, self.article_id)
 
     def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
         """
         Finds text of article
         """
-        pass
+        article_content = article_soup.find("div", class_="article__content")
+        text_paragraphs = article_content.find_all("p")
+        text = ''.join(i.text for i in text_paragraphs)
+        self.article.text = re.sub(r'\n+|\s{2,}', ' ', text)
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
-
-    def unify_date_format(self, date_str: str) -> datetime.datetime:
+        title = article_soup.find('h1', class_="article__title")
+        if title:
+            self.article.title = title.text
+        date = article_soup.find(class_="article__meta-date")
+        if date:
+            date_text = date.text
+            if not re.search(r'\d{4}', date_text):
+                curr_year = ' ' + str(datetime.date.today().year)
+                date_text = re.sub(r'(?<=[А-Яа-я])(?=,\s\d{2})', curr_year, date_text)
+            try:
+                self.unify_date_format(date_text)
+            except ValueError:
+                pass
+            else:
+                self.article.date = self.unify_date_format(date_text)
+        topics = [topic.text for topic in article_soup.find_all('a', class_="article-list__tag")]
+        self.article.author = ["NOT FOUND"]
+        if topics:
+            self.article.topics = topics
+
+    @staticmethod
+    def unify_date_format(date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        locale.setlocale(locale.LC_TIME, "ru_RU")
```

Comment:
using locale is possible but discouraged, as it has very dangerous outcomes for a larger program.  better to handle russian months in an explicit manner